### PR TITLE
Fail talisman if severity above threshold as defined in talismanRC(#114)

### PR DESCRIPTION
**Pull request by tt-official**
- Readme updated, see Configuring severity threshold section
- TalismanRC configuration for threshold severity. Default threshold is Low.
- Severity values range between 1(Low), 2(Medium), 3(High)
- pre-commit fails if eq or above threshold. If below threshold, only adds to warnings list
- custom patterns for detection are added as high severity patterns

### CODE REVIEW

**Comment by svishwanath-tw** [comment]
👍 . Great work on the PR.
I wonder if values like 1,2,3 can be ... more user friendly ?
Could we look for strings high, medium, low instead ?
What do others think @harinee @dineshba ?

**Comment by tt-official** [comment]
@svishwanath-tw I think it makes sense to make it more user-friendly. I'll update the MR accordingly.

File: detector/severity/pattern_severity.go [hunk 1]
```diff
@@ -0,0 +1,46 @@
+package severity
+
+import (
+	"regexp"
+)
+
+type SeverityValue int
+
+const (
+	LowSeverity    SeverityValue = 1
```
Comment (dineshba) [inline detector/severity/pattern_severity.go:10]: Can we use iota for this usecase ?

Comment (tt-official) [reply detector/severity/pattern_severity.go:10]: I'll make that change 👍

Comment (dineshba) [reply detector/severity/pattern_severity.go]: changes looks good. Thanks @tt-official  👍

**Comment by tt-official** [comment]
@svishwanath-tw @dineshba Can I merge this?

**Comment by svishwanath-tw** [comment]
@tt-official @dineshba I gave my approval long time ago. I will merge now.

**Pull request merged**